src/resample_functions.jl

```diff
@@ -0,0 +1,288 @@
+###
+
+### Resampler Functions
+
+###
+
+
+# Default resampling scheme
+
+function resample(w::AbstractVector{<:Real}, num_particles::Integer=length(w))
+
+    return resample_systematic(w, num_particles)
+
+end
+
+
+
+# More stable, faster version of rand(Categorical)
+
+function randcat(p::AbstractVector{T}) where T<:Real
+
+    r, s = rand(T), 1
+
+    for j in eachindex(p)
+
+        r -= p[j]
+
+        if r <= zero(T)
+
+            s = j
+
+            break
+
+        end
+
+    end
+
+    return s
+
+end
+
+
+
+function resample_multinomial(w::AbstractVector{<:Real}, num_particles::Integer)
+
+    return rand(Distributions.sampler(Categorical(w)), num_particles)
+
+end
+
+
+
+function resample_residual(w::AbstractVector{<:Real}, num_particles::Integer)
+
+
+
```

Comment:
I guess it would be nice to remove some of the white space?

Reply:
Yes, I am sorry. I have copied most of it from Turing. I will remove them!